examples/pln/johnny/Assertions.scm

```diff
@@ -0,0 +1,50 @@
+(ConceptNode "Johnny@444") (stv 0.99 1)
+
+(ConceptNode "Sweets@555") (stv 0.99 1)
+
+(ConceptNode "day@222") (stv 0.99 1)
+
+(ConceptNode "he@345") (stv 0.99 1)
+
+(ConceptNode "himself@456") (stv 0.99 1)
+
+(ConceptNode "risk@44") (stv 0.99 1)
+
+(ConceptNode "diabetes@66") (stv 0.99 1)
+
+(ImplicationLink (stv 0.99 1)
+	(EvaluationLink
+		(PredicateNode "every@111")
+        (ListLink
+			(ConceptNode "day@222")
+			(EvaluationLink
+				(PredicateNode "eating@333")
+				(ListLink
+					(ConceptNode "Johnny@444")
+					(ConceptNode "Sweets@555")))))
+	(EvaluationLink
+		(PredicateNode "at@123")
+		(ListLink
+			(EvaluationLink
+				(PredicateNode "placing@234")
+				(ListLink
+					(ConceptNode "he@345")
+					(ConceptNode "himself@456")))
+			(ConceptNode "risk@44"))))
+
+(EvaluationLink (stv 0.99 1)
+	(PredicateNode "for@55")
+	(ListLink
+		(ConceptNode "risk@44")
+		(ConceptNode "diabetes@66")))
+
+(NotLink (stv 0.99 1)
+	(EvaluationLink
+		(PredicateNode "every@111")
+        (ListLink
+			(ConceptNode "day@222")
+			(EvaluationLink
+				(PredicateNode "eating@333")
+				(ListLink
+					(ConceptNode "Johnny@444")
+					(ConceptNode "Sweets@555"))))))
```

Comment:
And . . . okay, I've fixed the SVIO rule and added a rule to take care of
that "if clause.  However, I can't seem to push those for the moment,
because I get a 403 web error when I do . . .

On Fri, May 29, 2015 at 4:25 PM, Aaron Nitzkin <a_nitzkin@yahoo.com> wrote:

> Okay, I've examined the output for this more carefully now.  Actually,
> it's almost all right. The two errors appear to be the interpreting of
> "risk" as an indirect object of placing, which is what you all wanted
> originally, but is an error in my mind, that is a result of one of the R2L
> rules for SVIO -- that one I can correct right away.  The other error is in
> what R2L is doing with the "if" which has to do with the last new rules I
> added, which were the first rules we had for such clauses, so that may take
> some fiddling . . .
>
> I don't think anything else is wrong with it.  It just looks odd, because
> it has all of the dependency relations treated separately, as we've been
> discussing -- either new post-processing or PLN is required to put them
> together into more wholistic structures.
>
> On Thu, May 28, 2015 at 6:14 AM, Linas Vepštas <notifications@github.com>
> wrote:
>
>> In examples/pln/johnny/Assertions.scm
>> <https://github.com/opencog/opencog/pull/1594#discussion_r31186337>:
>>
>> > +(EvaluationLink (stv 0.99 1)
>> > +	(PredicateNode "for@55")
>> > +	(ListLink
>> > +		(ConceptNode "risk@44")
>> > +		(ConceptNode "diabetes@66")))
>> > +
>> > +(NotLink (stv 0.99 1)
>> > +	(EvaluationLink
>> > +		(PredicateNode "every@111")
>> > +        (ListLink
>> > +			(ConceptNode "day@222")
>> > +			(EvaluationLink
>> > +				(PredicateNode "eating@333")
>> > +				(ListLink
>> > +					(ConceptNode "Johnny@444")
>> > +					(ConceptNode "Sweets@555"))))))
>>
>>  . . . It's like trying to put together one of those puzzles made of
>> wooden blocks, where as soon as you fit one piece in, three of the ones
>> that were in before, fall out . . .
>> Well, that is exactly the optimization problem I keep talking about. You
>> cannot just add on new rules, without also going back to modify the rules
>> you learned previously. The older rules were "correct" when dealing with
>> the smaller corpus, but, when enlarging your corpus, you find that some of
>> the old rules result in bad outcomes, bad deductions. You have to go back
>> and modify the old rules. This is like Ben's email where he asks "what N
>> can I choose for language" and the answer is you can't: its neither global
>> nor static. You worked with a small value of N at first, and learned some
>> confident rules for a limited corpus. As you start hearing, handling,
>> processing new classes of sentences, one of two things happen: either some
>> of the existing rules are confirmed, in which case your confidence of those
>> rules get *even bigger*, their status as bedrock rules is further
>> strengthened. The other possibility is that the existing rules get
>> modified. To find the right modification, you have to go to deeper bedrock,
>> and solve the global problem of (deeper_bedrock + modified_rules +
>> new_corpus) is minimally conflicted, inconsistent. This makes the learning
>> process hierarchical, and buries some rules deeply, where they can't be
>> modified without tearing down the whole edifice. Other rules remain
>> superficial and malleable, and don't harden into a fixed form until a lot
>> of corpus experience is built up.
>> … <#14d9ec6e7cbe973e_14d9770afd58faa7_>
>> -- Linas
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/opencog/opencog/pull/1594/files#r31186337>.
>>
>
>